### PR TITLE
Aribtrary filesystem writes in tar 0.4.15 and older

### PR DESCRIPTION
**Pull request by alexcrichton**
More details inside!

### CODE REVIEW

File: crates/tar/RUSTSEC-2018-0002.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+with hard links or symlinks, however, can be used to overwrite any file on the
+filesystem.
+
+Tarballs can contain multiple entries for the same file. A tarball which first
+contains an entry for a hard link or symlink pointing to any file on the
+filesystem will have the link created, and then afterwards if the same file is
+listed in the tarball the hard link will be rewritten and any file can be
+rewritten on the filesystem.
+
+This has been fixed in https://github.com/alexcrichton/tar/pull/156 and is
```
Comment (niklasf) [inline crates/tar/RUSTSEC-2018-0002.toml]: s/tar/tar-rs/

**Comment by tirkarthi** [comment]
I think this is towards the snyk report. Relevant zip-rs issue : https://github.com/mvdnes/zip-rs/issues/74

Ref : https://github.com/snyk/zip-slip-vulnerability

**Comment by tarcieri** [comment]
@greysteil yes, however given the current way the `rustsec-client` crate works, the rest of the advisories are combined into `Advisories.toml` in the root of the repo, which is an array of advisories. The other TOML files aren't consumed directly.

I'm traveling right now but I'll fix it up and get it into `Advisories.toml` shortly.

(As a sidebar I should probably just get rid of `Advisories.toml` and consume this repo directly via a git crate)

**Comment by greysteil** [comment]
👍 - I'm pulling in the individual files in Dependabot, but have updated my processing to handle formatting like this, too.

Enjoy your travels!

**Pull request merged**